tests/test_category.py

```diff
@@ -0,0 +1,116 @@
+        Category.objects.create(name='tomatos category', owner=self.user)
+
+        self.assertTrue(self.client.login(**self.user_info))
+        response = self.client.get('/category/')
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(json.loads(response.content)['count'], 2)
+
+        serializer = CategorySerializer(
+            Category.objects.filter(
+                Q(owner__isnull=True) |
+                Q(owner=self.user)).order_by('-created_at'),
+            many=True)
+        serializer_data = json.loads(json.dumps(serializer.data))
+        response_data = json.loads(response.content)['results']
+
+        # check the fields
+        self.assertListEqual(
+            list(response_data[0].keys()),
+            ['id', 'parent', 'name', 'owner'])
+
+        # check the data
+        self.assertEqual(serializer_data, response_data)
+
+    def test_create_anonymousUser(self):
+        response = self.client.post('/category/', self.cdict, format='json')
+        self.assertEqual(response.status_code, 403)
+
+    def test_create_empty_name(self):
+        # test category name can't be empty
+        self.assertTrue(self.client.login(**self.user_info))
+        response = self.client.post('/category/', {'name': ''}, format='json')
+        self.assertEqual(response.status_code, 400)
+
+        response = self.client.post('/category/', {'name': None}, format='json')
+        self.assertEqual(response.status_code, 400)
+
+    def test_create_no_parent(self):
+        # test create category with no parent
+        self.assertTrue(self.client.login(**self.user_info))
+        response = self.client.post('/category/', self.cdict, format='json')
+        self.assertEqual(response.status_code, 201)
+        serializer = CategorySerializer(Category.objects.get(owner=self.user, name=self.cname))
+        self.assertEqual(
+            json.loads(json.dumps(serializer.data)),
+            json.loads(response.content))
+
+    def test_create_with_parent(self):
+        self.assertTrue(self.client.login(**self.user_info))
+        # test create category with parent
+        parent = Category.objects.create(owner=self.user, name='papa')
+        response = self.client.post('/category/', {'name': self.cname, 'parent_id': parent.id}, format='json')
+        self.assertEqual(response.status_code, 201)
+        serializer = CategorySerializer(Category.objects.get(owner=self.user, name=self.cname))
+        self.assertEqual(
+            json.loads(json.dumps(serializer.data)),
+            json.loads(response.content))
+
+        # check all fof the fields needed in the results.
+        self.assertListEqual(
+            list(json.loads(response.content).keys()),
+            ['id', 'parent', 'name', 'owner'])
+
+    def test_retrieve(self):
+        pass
+
+    def test_update(self):
+        pass
+
+    def test_delete(self):
+        pass
+
+    def test_model_duplicate(self):
```

Comment:
You don't have to test everything related to category in a single TestCase. Create another one for models.